parameter/CompoundRule.cpp

```diff
@@ -140,7 +140,9 @@ bool CCompoundRule::matches() const
 bool CCompoundRule::fromXml(const CXmlElement& xmlElement, CXmlSerializingContext& serializingContext)
 {
     // Get type
-    _bTypeAll = xmlElement.getAttributeBoolean("Type", _apcTypes[true]);
+    string strType;
+    xmlElement.getAttribute("Type", strType);
+    _bTypeAll = strType == _apcTypes[true];
```

Comment:
What value should be returned if getAttribute fails ?

If it returns the value of the default constructor, this leads to:

```
bool error;
string strType = xmlElement.getAttribute("Type", &error);
_bTypeAll = strType == _apcTypes[true];
```

is it so much nicer ?